### PR TITLE
Reuse Symfony unstable workflow

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,11 +21,10 @@
     ],
     "require": {
         "php": "^7.2|^8.0",
-        "symfony/framework-bundle": "~3.4|~4.0|~5.0|~6.0",
```
Comment (derrabus) [inline composer.json]: Should be reverted. Otherwise Composer would refuse to install the bundle with Symfony 6 again.

File: psalm.xml [hunk 2]
```diff
@@ -15,13 +15,4 @@
...
     </projectFiles>
-
-    <issueHandlers>
-        <!-- This check is confused by Symfony's "mixed" property types. -->
-        <ReservedWord>
-            <errorLevel type="suppress">
-                <directory name="." />
-            </errorLevel>
-        </ReservedWord>
-    </issueHandlers>
```
Comment (derrabus) [inline psalm.xml]: As soon as Symfony 6 is stable, Psalm will fail again. Let's keep the issue handler in here until we have a better solution.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (greg0ire) [reply composer.json]: It's just here to demonstrate how it's possible to test versions that are not supported officially. Will be reverted 👍

File: psalm.xml (hunk 2 above)
Comment (greg0ire) [reply psalm.xml]: I thought this was because we were using bleeding edge Psalm?

Comment (derrabus) [reply psalm.xml]: I thought this was a bleeding edge Psalm issue at first, but it turns out that stable Psalm has the same issue. :-/

**Comment by greg0ire** [comment]
@derrabus I just pushed with `6.0.x-dev` as a constraint, but this only results in the framework-bundle being installed in that version. I think the case of Symfony is kind of special. Is there a trick I could use in the workflow to have all symfony components on 6.0.x-dev, and all other packages in a stable version?

**Comment by derrabus** [comment]
> Is there a trick I could use in the workflow to have all symfony components on 6.0.x-dev, and all other packages in a stable version?

If you add `flex` to the `tools` parameter of the Setup PHP step, this should do the trick:

```sh
composer config extra.symfony.require '6.0.*'
```

But I just noticed that `doctrine/migrations` is blocking `symfony/console` 6. We need to solve that first, I guess.

**Comment by greg0ire** [comment]
Then I should make the upstream workflow Symfony-specific, right? This wouldn't work for laminas, would it? Another solution would be to require `symfony/symfony 6.0.x`, but I guess cloning the whole repo might be slow. Let me try.

**Comment by greg0ire** [comment]
That does not work, because I would have to drop `symfony/framework-bundle`

**Comment by greg0ire** [comment]
@derrabus I think I have managed to do something that behaves correctly (as in it allows us to detect that we should focus on `doctrine/migrations` first)

**Comment by greg0ire** [comment]
ping @derrabus :slightly_smiling_face:

**Comment by derrabus** [comment]
On the Symfony 6 workflow, Composer installs the `symfony/symfony` monorepo now which is certainly not what we want. I wonder why that happens.

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -15,3 +15,10 @@ jobs:
     uses: "doctrine/.github/.github/workflows/continuous-integration.yml@1.2.1"
     with:
       php-versions: '["7.2", "7.3", "7.4", "8.0"]'
+
+  phpunit-with-symfony-6:
+    name: "PHPUnit with Symfony 6.0.x-dev"
+    uses: "doctrine/.github/.github/workflows/continuous-integration-symfony-unstable.yml@1.4.0"
+    with:
+      symfony-version-constraint: "6.0.x-dev"
+      extra-requirements: "symfony/framework-bundle 6.0.x-dev symfony/symfony 6.0.x-dev"
```
Comment (derrabus) [inline .github/workflows/continuous-integration.yml]:
```suggestion
      symfony-version-constraint: "6.0.*"
```

Can we try this?

**Comment by greg0ire** [comment]
> On the Symfony 6 workflow, Composer installs the symfony/symfony monorepo now which is certainly not what we want. I wonder why that happens. 

Well spotted! And also… WTF :sweat_smile: ? 

EDIT: Oh… it's just because I wrote `symfony/symfony` in the requirements :facepalm: … a remnant of my previous attempts. Anyway, as you pointed out, using extra requirements is not needed here since we are allowing Symfony 6 already.

**Pull request merged**